fixieai/agents/api.py

```diff
@@ -16,10 +18,31 @@ class Embed:
     # The MIME content type of the object, e.g., "image/png" or "application/json".
     content_type: str
 
-    # A public URL where the object can be downloaded. The Embed API can be used to
-    # upload an Embed object to Fixie and generate a URL.
+    # A public URL where the object can be downloaded. This can be a data URI.
     uri: str
 
+    @property
+    def content(self) -> bytes:
+        """Retrieves the content for this Embed object."""
+        if self.uri.startswith("data:"):
```

Comment:
consider using a standard url format like: "base64://bytes" so we can parse with [urlllib.parse.urlparse](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse)

Reply:
AFAIK `base64` is not an official scheme, but `data` is: https://www.rfc-editor.org/rfc/rfc2397